### ISSUE TITLE
Uncaught exception when adding a specific breakpoint in text-editor.js

### ISSUE THREAD

**Issue by Ben3eeE** [atom/atom#11449]
### Prerequisites

* [X] Can you reproduce the problem in [safe mode](http://flight-manual.atom.io/hacking-atom/sections/debugging/#check-if-the-problem-shows-up-in-safe-mode)?
* [X] Are you running the [latest version of Atom](http://flight-manual.atom.io/hacking-atom/sections/debugging/#update-to-the-latest-version)?
* [X] Did you check the [debugging guide](http://flight-manual.atom.io/hacking-atom/sections/debugging/)?
* [X] Did you check the [FAQs on Discuss](https://discuss.atom.io/c/faq)?
* [X] Are you reporting to the [correct repository](https://github.com/atom/atom/blob/master/CONTRIBUTING.md#atom-and-packages)?
* [X] Did you [perform a cursory search](https://github.com/issues?q=is%3Aissue+user%3Aatom+-repo%3Aatom%2Felectron) to see if your bug or enhancement is already reported?

For more information on how to write a good [bug report](https://github.com/atom/atom/blob/master/CONTRIBUTING.md#how-do-i-submit-a-good-bug-report) or [enhancement request](https://github.com/atom/atom/blob/master/CONTRIBUTING.md#how-do-i-submit-a-good-enhancement-suggestion), see the `CONTRIBUTING` guide.

### Description

Stumbled upon https://github.com/atom/atom/issues/9929 by accident while stepping through text-editor.js code in the developer tools. Was able to find repro steps that work every time.
Is stepping through random parts of atom source(from app.asar) not supported and will cause random things like this? I have no special tools from atom just the editor.
Was talking with @lee-dohm on the atom slack channel and he recommended I open a new issue for this.

### Steps to Reproduce

1. Open atom in safe mode.
2. Ensure you have Tab Type set to auto.
3. Open a new tab and type some text.
4. Open a new empty tab.
4. Put a breakpoint in the function `TextEditor.prototype.usesSoftTabs` in text-editor.js. I put it on the first line of the function which is line 2261.
6. Split the empty tab down.
7. Atom is now "soft locked", can't add text to any of the open tabs while other parts of the UI work fine.
8. Switch to tab with text from step 3, tab is empty no text visible.
9. Uncaught exception when clicking in the tab.

What is strange is that execution doesn't stop on the breakpoint when doing split down.
The execution does however stop on the breakpoint on some other actions. For example new tab stops while split tab doesn't stop.
Also just putting a breakpoint on this line and splitting causes problems with syntax coloring being gone and other random things. The exception seems to just be a special case of this.

**Expected behavior:** Atom to not throw an exception and stop working, perhaps to break on the breakpoint I added.

**Actual behavior:**

**Atom Version**: 1.6.2
**System**: Microsoft Windows 7 Professional 
**Thrown From**: Atom Core

### Stack Trace

Uncaught Error: Undefined screen line when clipping screen position

```
At C:\Users\bene\AppData\Local\atom\app-1.6.2\resources\app.asar\src\display-buffer.js:886

Error: Undefined screen line when clipping screen position
    at DisplayBuffer.module.exports.DisplayBuffer.clipScreenPosition (C:\Users\bene\AppData\Local\atom\app-1.6.2\resources\app.asar\src\display-buffer.js:874:15)
    at DisplayBuffer.module.exports.DisplayBuffer.bufferPositionForScreenPosition (C:\Users\bene\AppData\Local\atom\app-1.6.2\resources\app.asar\src\display-buffer.js:826:20)
    at TextEditorMarker.module.exports.TextEditorMarker.setHeadScreenPosition (C:\Users\bene\AppData\Local\atom\app-1.6.2\resources\app.asar\src\text-editor-marker.js:185:60)
    at C:\Users\bene\AppData\Local\atom\app-1.6.2\resources\app.asar\src\cursor.js:67:31
    at Cursor.module.exports.Cursor.changePosition (C:\Users\bene\AppData\Local\atom\app-1.6.2\resources\app.asar\src\cursor.js:673:7)
    at Cursor.module.exports.Cursor.setScreenPosition (C:\Users\bene\AppData\Local\atom\app-1.6.2\resources\app.asar\src\cursor.js:65:19)
    at C:\Users\bene\AppData\Local\atom\app-1.6.2\resources\app.asar\src\text-editor.js:1475:23
    at TextEditor.module.exports.TextEditor.moveCursors (C:\Users\bene\AppData\Local\atom\app-1.6.2\resources\app.asar\src\text-editor.js:1679:9)
    at TextEditor.module.exports.TextEditor.setCursorScreenPosition (C:\Users\bene\AppData\Local\atom\app-1.6.2\resources\app.asar\src\text-editor.js:1474:19)
    at TextEditorComponent.module.exports.TextEditorComponent.onMouseDown (C:\Users\bene\AppData\Local\atom\app-1.6.2\resources\app.asar\src\text-editor-component.js:729:25)
    at HTMLDivElement.<anonymous> (C:\Users\bene\AppData\Local\atom\app-1.6.2\resources\app.asar\src\text-editor-component.js:3:61)
    at EventEmitter.ipc.sendSync (C:\Users\bene\AppData\Local\atom\app-1.6.2\resources\atom.asar\renderer\api\lib\ipc.js:21:31)
    at BrowserWindow.RemoteMemberFunction [as emit] (C:\Users\bene\AppData\Local\atom\app-1.6.2\resources\atom.asar\renderer\api\lib\remote.js:143:29)
    at ContextMenuManager.module.exports.ContextMenuManager.showForEvent (C:\Users\bene\AppData\Local\atom\app-1.6.2\resources\app.asar\src\context-menu-manager.js:172:33)
    at WindowEventHandler.module.exports.WindowEventHandler.handleDocumentContextmenu (C:\Users\bene\AppData\Local\atom\app-1.6.2\resources\app.asar\src\window-event-handler.js:250:47)
    at HTMLDocument.<anonymous> (C:\Users\bene\AppData\Local\atom\app-1.6.2\resources\app.asar\src\window-event-handler.js:3:61)
```

### Commands

```
     -0:18.9.0 window:toggle-dev-tools (atom-text-editor.editor.is-focused)
     -0:12 tabs:split-down (div.title)
```

### Config

```json
{
  "core": {
    "autoHideMenuBar": true,
    "disabledPackages": [
      "linter-pep8",
      "about"
    ],
    "themes": [
      "atom-dark-ui",
      "atom-dark-syntax"
    ]
  },
  "editor": {
    "fontSize": 13,
    "invisibles": {},
    "scrollPastEnd": true,
    "showIndentGuide": true
  }
}
```

### Installed Packages

```coffee
# User
No installed packages

# Dev
No dev packages
```

### Versions

Reproduced both in Windows 7 and Ubuntu 15.10 VM.

Atom 1.6.2.
Ubuntu 15.10 in Oracle VM Windows 7 host.
apm 1.6.0
npm 2.13.3
node 0.10.40
python 2.7.10
git 2.5.0

Atom 1.6.2
Windows 7
apm  1.6.0
npm  2.13.3
node 0.10.40
python 2.7.11
git 1.9.5.msysgit.1
visual studio 2012

**Comment by Ben3eeE** [comment]
Update

In atom 1.7.0 I can no longer reproduce this and execution stops on the breakpoint when splitting panes as expected. Maybe this was a bug in the developer tools that was fixed with the update? Maybe there is some flaky code in atom before this function that causes the error that was changed with the update? Maybe the butterflies in the universe realigned with the release of 1.7.0 and the magical breakpoint is supposed to be at another place in the code.
In atom 1.6.2 I was able to reproduce it all times on multiple computers/oses.

**Comment by 50Wliu** [comment]
Atom 1.7.0 has an updated Electron version which probably contains an updated dev tools as well, so if this is fixed in 1.7.0, :sparkles:?

**Comment by Ben3eeE** [comment]
I guess this can be closed, unless someone (like @as-cii he likes to hear about weird things) wants to dig to be sure that the electron update fixed this. What is interesting maybe is that the description others have of #9929 is exactly the same as I felt when doing this. So while the steps to reproduce might be different the state we end up in feels the same and maybe gives clue to the other, or something.
Can someone reproduce this with the steps provided? Does anyone feel we should care? Should we maybe inform upstream since maybe it is a dev tools issue and maybe it wasn't actually fixed just changed with update, it actually feels extremely volatile, my testing revealed that the exact line of the breakpoint was important, I was only able to reproduce this reliably with the breakpoint on the first or the second line of this function while others worked fine.

Just random thoughts I have before closing this.
